src/main.cpp

```diff
@@ -5340,26 +5350,20 @@ bool SendMessages(CNode* pto, bool fSendTrickle)
 
     // Address refresh broadcast
     static int64_t nLastRebroadcast;
-    if (!IsInitialBlockDownload() && ( GetAdjustedTime() - nLastRebroadcast > 24 * 60 * 60))
+    if (!IsInitialBlockDownload() && ( GetAdjustedTime() - nLastRebroadcast > 1 * 60 * 60))
```

Comment:
In `AdvertiseLocal` is currently the only place a new IP could be chosen to be pushed, but only every 24, and only by chance.
My ISP forces me to use a new IP after 24 hours, so the interval should be reduced, but if used on servers with static IP it might be spam. what are your opinions?
Currently there is also a 2h interval that forces disconnects in `net.cpp:1203` that is marked as `consider for removal` but i think it could help to keep a dynamic in the network and prevent it to split.
So also the 2h interval is an argument to reduce this advertisement interval.

Reply:
@Tetrix42 Thanks for your analysis. Your comments are correct. Bitcoin's approach is passive and mostly depends on a node's outbound connections to propagate dynamic IP address reassignments. When your node connects to another listening node, that peer records your current IP address and then&mdash;through peer exchange&mdash;sends it to other nodes. Of course, that only works when your node pushes back the IP address that the peer reported in `addrLocal`&mdash;it won't work very well when your node sends the IP address assigned by UPnP after that IP address changes.

As a result, the current "visible" IP address of a listening node with dynamic reassignment is often different from the inventory of IP addresses that the node detected for itself during startup via `AddLocal()`. This interaction is less reactive to IP address changes than the direct approach that you originally proposed, but it reduces overhead, mutable state, and attack surface by relying the existing peer discovery protocol. 

Personally, I think it's a rather messy way to solve the problem. We can improve this area after we merge Bitcoin's newer peer-to-peer networking code, including some of your original ideas. Since most listening nodes bind to IP addresses that change very infrequently, I'm not sure that it's worth introducing extra overhead to handle this at the moment. Perhaps we can make the interval for `AdvertiseLocal()` configurable (with a default at 24 hours) for nodes that need to push updates more frequently. 

> Currently there is also a 2h interval that forces disconnects in net.cpp:1203 that is marked as consider for removal but i think it could help to keep a dynamic in the network and prevent it to split.

I agree&mdash;the connection cycling is a nice property for decentralization. However, it also makes it easier for an actor to gradually take control of inbound connections on listening nodes to partition the network or execute a DoS attack. Bitcoin has more advanced routines now for connection eviction that attempt to balance these concerns...another reason to update our code base.